locksmith/src/controllers/v2/priceController.ts

```diff
@@ -20,10 +16,20 @@ export const amount: RequestHandler = async (request, response) => {
     ? erc20Address
     : undefined
 
-  const result = await defiLammaPrice({
+  const _lockAddress = request.query.lockAddress?.toString()
+  const keysToPurchase = Number(request.query.keysToPurchase?.toString() || 1)
```

Comment:
I'm confused. Isn't this what amount is for?

Reply:
`amount` is the amount related to the lock price eg . `50 USDC` has `50` as amount, and this is used to get the correct price conversion 

`keysToPurchase` in the numbers of keys we are purchasing, is needed to multiply the custom credit card price to have the correct final price

Reply:
@kalidiagne Please let's use a better name for that function. I agree with @searchableguy that this is unclear.